src/imgtools/dicom/crawl/parse_dicoms.py

```
@@ -466,3 +470,57 @@
+        list[dict[str, str]]
+            - the barebones dict with duplicates removed.    
+
+    Note: I tried to make this as efficient as possible, sorry if it slows down everything. 
+    """
+
+    hash_values = set()
+    # I find it more intuitive to pass in a list, rather than a set, but since sets are faster for our usecase I cast ignore_keys to a set.
+    ignore_keys = set(ignore_keys) 
+    output = []
+    for record in slim_db:
+        hash_key = [record[key] for key in record.keys() if key not in ignore_keys]
```

Ruff (0.11.9)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)